### PR TITLE
Leave PR comments without `workflow_dispatch`

### PR DESCRIPTION
**Pull request by swissspidy**
Trac ticket: 

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

File: .github/workflows/pull-request-comments.yml [hunk 1]
```diff
@@ -87,17 +85,47 @@ jobs:
...
+            var download = await github.actions.downloadArtifact({
+               owner: context.repo.owner,
+               repo: context.repo.repo,
+               artifact_id: matchArtifact.id,
+               archive_format: 'zip',
+            });
+            var fs = require('fs');
+            fs.writeFileSync('${{github.workspace}}/pr.zip', Buffer.from(download.data));
+
+      - run: unzip pr.zip
```
Comment (desrosj) [inline .github/workflows/pull-request-comments.yml]: Can we add `name:`s to each step? A few are missing.

File: .github/workflows/test-build-processes.yml [hunk 2]
```diff
@@ -117,21 +118,14 @@ jobs:
...
-            github.rest.actions.createWorkflowDispatch({
-              owner: context.repo.owner,
-              repo: context.repo.repo,
-              workflow_id: 'pull-request-comments.yml',
-              ref: 'trunk',
-              inputs: {
-                pr_number: '${{ github.event.number }}'
-              }
-            });
+          name: pr
```
Comment (desrosj) [inline .github/workflows/test-build-processes.yml]: I know this is the example, but could we make it more specific? `pr-number` maybe?

**Comment by adamziel** [comment]
I love this idea! Unfortunately I couldn't get it to work on my fork, no matter what I did. I got down to this barebones workflow file, and it still wouldn't run:

```
name: Pull Request Comments

on:
  workflow_run:
    workflows:
      - Test Build Processes
    types:
      - completed

jobs:
  test-job:
    name: Test Step
    runs-on: ubuntu-latest
    steps:
      - run: env
```

I suspect this might be related to this note in the workflows documentation:

> When you use the repository's GITHUB_TOKEN to perform tasks, events triggered by the GITHUB_TOKEN, with the exception of workflow_dispatch and repository_dispatch, will not create a new workflow run.

https://docs.github.com/en/actions/using-workflows/triggering-a-workflow#triggering-a-workflow-from-a-workflow

For this reason, https://github.com/WordPress/wordpress-develop/pull/5737 might be the best shot at this. Either that, or creating that comment in the same workflow file that runs the builds.

**Comment by swissspidy** [comment]
I got it working successfully over at my fork, see https://github.com/swissspidy/wordpress-develop/pull/55

According to https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows, the `workflow_run` event will only trigger a workflow run if the workflow file is on the default branch. So once I added it to `trunk` on my fork, it worked. So similar to `pull_request_target` where the file needs to exist on the target branch as well.

So `workflow_run` definitely works.

File: .github/workflows/pull-request-comments.yml [hunk 3]
```diff
@@ -81,17 +83,55 @@ jobs:
     permissions:
       issues: write
       pull-requests: write
-    if: ${{ github.repository == 'WordPress/wordpress-develop' && github.event_name == 'pull_request_target' }}
+    if: >
+      github.repository == 'WordPress/wordpress-develop' &&
+      github.event.workflow_run.event == 'pull_request' &&
+      github.event.workflow_run.conclusion == 'success'
     steps:
+      - name: 'Download artifact'
```
Comment (desrosj) [inline .github/workflows/pull-request-comments.yml]:
```suggestion
      - name: Download artifact
```

File: .github/workflows/pull-request-comments.yml [hunk 4]
```diff
@@ -4,8 +4,10 @@ name: Pull Request Comments
 on:
   pull_request_target:
     types: [ 'opened' ]
-    branches:
-      - trunk
+  workflow_run:
+    workflows: ["Test Build Processes"]
```
Comment (desrosj) [inline .github/workflows/pull-request-comments.yml]:
```suggestion
    workflows: [ 'Test Build Processes' ]
```

**Review by desrosj — APPROVED** [review]
Sorry! I had started a PR review and never hit submit.

Rechecked my comments and made a few additional suggestions, none of which I feel are really blockers. Let's give it a shot!

**Comment by swissspidy** [comment]
Committed in https://core.trac.wordpress.org/changeset/57210

**Pull request closed without merging**